### PR TITLE
test: add selecting resource related e2e tests

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

add basic selecting resource related e2e tests

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### Special notes for your reviewer

more tests will be added in the next PR

### CODE REVIEW

**Review by ryanzhang-oss — APPROVED** [review]

**Comment by zhiying-lin** [comment]
need to rebase code to the master, will sent out the new commit, thanks

File: test/e2e/actuals_test.go [hunk 1]
```diff
@@ -202,21 +202,14 @@ func crpStatusUpdatedActual() func() error {
 	}
 }
 
-func workNamespaceAndDeploymentRemovedFromClusterActual(cluster *framework.Cluster) func() error {
+func workNamespaceRemovedFromClusterActual(cluster *framework.Cluster) func() error {
 	client := cluster.KubeClient
 
 	workNamespaceName := fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess())
-	appConfigMapName := fmt.Sprintf(appConfigMapNameTemplate, GinkgoParallelProcess())
```
Comment (michaelawyu) [inline test/e2e/actuals_test.go:209]: The placement test uses the app config I believe?

Comment (michaelawyu) [reply test/e2e/actuals_test.go:209]: That logic is thus missing in many of the parts below I believe.

Comment (zhiying-lin) [reply test/e2e/actuals_test.go:209]: app config will be created in the work namespace. So there is no need to check if app config is there if work namespace has been deleted.

Comment (michaelawyu) [reply test/e2e/actuals_test.go:209]: Oh, this is the removed actual. I read it wrong.

**Review by michaelawyu — APPROVED** [review]
LGTM.

**Pull request merged**